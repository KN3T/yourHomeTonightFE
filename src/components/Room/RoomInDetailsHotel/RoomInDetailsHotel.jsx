/* eslint-disable react/prop-types */
import { CaretRightOutlined } from '@ant-design/icons';
import { Button, Col, Image, Row, Space } from 'antd';
import React from 'react';
import { useTranslation } from 'react-i18next';

import formatCurrency from '../../../utils/formatCurrency';
import './index.scss';

const RoomInDetailsHotel = ({ room, showModal }) => {
  const { t, i18n } = useTranslation();
  const currentLanguage = i18n.language;

  return (
    <Row gutter={[20, 20]}>
      <Col span={6}>
        <span>{room.type}</span>
      </Col>
      <Col span={6}>
        <Space direction="vertical">
          {room.asset.slice(0, 3).map((asset, key) => (
            <span style={{ color: '#148A57' }} key={key}>
              {asset}
            </span>
          ))}
        </Space>
      </Col>

      <Col span={6} style={{ display: 'flex', justifyContent: 'center' }}>
        <span className="price" style={{ fontSize: '20px' }}>
          <b>
            {t('details__hotel.price_value', {
              val: formatCurrency(room.price, currentLanguage),
            })}
          </b>
        </span>
      </Col>
      <Col span={6} style={{ display: 'flex', justifyContent: 'center' }}>
        <Button
          onClick={() => showModal(room.id)}
          icon={<CaretRightOutlined />}
          type="primary"
        >
          {t('details__hotel.view_details')}
        </Button>
      </Col>
    </Row>
  );
};

export default RoomInDetailsHotel;
