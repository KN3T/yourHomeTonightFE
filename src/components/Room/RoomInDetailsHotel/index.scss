.room__in__details__hotel {
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #6f688d;
  .price {
    font-weight: bold;
    font-size: 18px;
    color: black;
  }
}
