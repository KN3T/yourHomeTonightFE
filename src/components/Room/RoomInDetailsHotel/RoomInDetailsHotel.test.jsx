import { render } from '@testing-library/react';
import React from 'react';
import { describe } from 'vitest';

import RoomInDetailsHotel from './RoomInDetailsHotel';

describe('RoomInDetailsHotel', () => {
  it('Should return an item room in table', () => {
    const room = {
      adults: 3,
      asset: ['Dishwasher', 'Flat-screen TV'],
      beds: 1,
      children: 1,
      description: 'This triple room has a balcony, sofa and minibar.',
      id: 20,
      images: [
        { imageId: 112, src: 'URL1' },
        { imageId: 113, src: 'URL2' },
        { imageId: 114, src: 'URL3' },
      ],
      number: 6,
      price: 36,
      rating: 0,
      type: 'Deluxe Triple Room',
    };
    const showRoom = () => console.log('Show room ID');
    const roomInDetailsHotel = render(
      <RoomInDetailsHotel room={room} showRoom={showRoom} />
    );
    expect(roomInDetailsHotel).toMatchSnapshot();
  });
});
