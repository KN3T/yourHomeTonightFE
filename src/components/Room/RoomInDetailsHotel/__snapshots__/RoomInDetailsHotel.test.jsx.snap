// Vitest Snapshot v1

exports[`RoomInDetailsHotel > Should return an item room in table 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-row"
        style="margin: -10px -10px -10px -10px;"
      >
        <div
          class="ant-col ant-col-6"
          style="padding: 10px 10px 10px 10px;"
        >
          <span>
            Deluxe Triple Room
          </span>
        </div>
        <div
          class="ant-col ant-col-6"
          style="padding: 10px 10px 10px 10px;"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 8px;"
            >
              <span
                style="color: rgb(20, 138, 87);"
              >
                Dishwasher
              </span>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                style="color: rgb(20, 138, 87);"
              >
                Flat-screen TV
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-6"
          style="padding: 10px 10px 10px 10px; display: flex; justify-content: center;"
        >
          <span
            class="price"
            style="font-size: 20px;"
          >
            <b>
              details__hotel.price_value
            </b>
          </span>
        </div>
        <div
          class="ant-col ant-col-6"
          style="padding: 10px 10px 10px 10px; display: flex; justify-content: center;"
        >
          <button
            class="ant-btn ant-btn-primary"
            data-testid="button-view-detail"
            type="button"
          >
            <span
              aria-label="caret-right"
              class="anticon anticon-caret-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="caret-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                />
              </svg>
            </span>
            <span>
              details__hotel.view_details
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-row"
      style="margin: -10px -10px -10px -10px;"
    >
      <div
        class="ant-col ant-col-6"
        style="padding: 10px 10px 10px 10px;"
      >
        <span>
          Deluxe Triple Room
        </span>
      </div>
      <div
        class="ant-col ant-col-6"
        style="padding: 10px 10px 10px 10px;"
      >
        <div
          class="ant-space ant-space-vertical"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <span
              style="color: rgb(20, 138, 87);"
            >
              Dishwasher
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <span
              style="color: rgb(20, 138, 87);"
            >
              Flat-screen TV
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-6"
        style="padding: 10px 10px 10px 10px; display: flex; justify-content: center;"
      >
        <span
          class="price"
          style="font-size: 20px;"
        >
          <b>
            details__hotel.price_value
          </b>
        </span>
      </div>
      <div
        class="ant-col ant-col-6"
        style="padding: 10px 10px 10px 10px; display: flex; justify-content: center;"
      >
        <button
          class="ant-btn ant-btn-primary"
          data-testid="button-view-detail"
          type="button"
        >
          <span
            aria-label="caret-right"
            class="anticon anticon-caret-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="caret-right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
              />
            </svg>
          </span>
          <span>
            details__hotel.view_details
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
